### PR TITLE
remove __pillar__ refs

### LINKED ISSUE #93835: Salt / _PILLAR_ references in DNS parts of the codebase - what's the plan to remove?
**What happened**:

There are a lot of `_PILLAR_` variables (i.e. around the DNS area) which need to probably be removed over time, as I think were not coupled to salt anymore. 

**What you expected to happen**:

Salt variable semantics wouldn't be coupled into the codebase.

**How to reproduce it (as minimally and precisely as possible)**:

(`grep -I` for `_pillar_` in the codebase) 

**Environment**:
- Kubernetes version: any

### PR DESCRIPTION
**Pull request by jayunit100**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

salt cleanup

**Which issue(s) this PR fixes**:
Fixes #93835

### CODE REVIEW

**Comment by jayunit100** [comment]
```



[sig-scheduling] SchedulerPreemption [Serial] validates basic preemption works [Conformance] expand_more1m0s | [sig-scheduling] SchedulerPreemption [Serial] validates basic preemption works [Conformance] expand_more | 1m0s
[sig-scheduling] SchedulerPreemption [Serial] validates basic preemption works [Conformance] expand_more | 1m0s
[sig-scheduling] SchedulerPreemption [Serial] validates lower priority pod preemption by critical pod [Conformance] expand_more1m0s | [sig-scheduling] SchedulerPreemption [Serial] validates lower priority pod preemption by critical pod [Conformance] expand_more | 1m0s
[sig-scheduling] SchedulerPreemption [Serial] validates lower priority pod preemption by critical pod [Conformance] expand_more | 1m0s
```

Scheduler tests failed, seems unrelated, retrying..

**Comment by jayunit100** [comment]
no idea how I closed this, reopening.

File: cluster/addons/dns/nodelocaldns/nodelocaldns.yaml [hunk 1]
```diff
@@ -96,8 +96,8 @@ data:
         cache 30
         reload
         loop
-        bind __PILLAR__LOCAL__DNS__ __PILLAR__DNS__SERVER__
-        forward . __PILLAR__UPSTREAM__SERVERS__ {
+        bind __LOCAL__DNS__ __DNS__SERVER__
```
Comment (prameshj) [inline cluster/addons/dns/nodelocaldns/nodelocaldns.yaml:99]: node-local-dns binary depends on this string - https://github.com/kubernetes/dns/blob/69830ba1a789b3e44d4bb37a96ef01332b8044a0/cmd/node-cache/app/configmap.go#L26

it substitutes __PILLAR__DNS__SERVER__ as the kube-dns service IP, if it is not already set. Is it ok to leave this change out or include the change in node-local-dns as well?

Comment (jayunit100) [reply cluster/addons/dns/nodelocaldns/nodelocaldns.yaml:99]: ah ok gotcha, ill update that file as well thanks

Comment (jayunit100) [reply cluster/addons/dns/nodelocaldns/nodelocaldns.yaml:99]: https://github.com/kubernetes/dns/pull/395 <-- assuming that should be sufficient to fix it ?

Comment (jayunit100) [reply cluster/addons/dns/nodelocaldns/nodelocaldns.yaml:99]: resolving, I assume the companion PR now resolves this

**Comment by jayunit100** [comment]
https://github.com/kubernetes/dns/pull/395 <--- @prameshj , lmk if that companion PR works, resolved the other comment.

**Comment by prameshj** [comment]
> [kubernetes/dns#395](https://github.com/kubernetes/dns/pull/395) <--- @prameshj , lmk if that companion PR works, resolved the other comment.

Just replacing with the new string could break deployments that are updating the node-cache image but not updating the configmap definition. The configs could be picked up from other places than just the addons/dns/nodelocaldns. We could add logic to replace both the *PILLAR_* strings as well as the new strings, in node-cache. I am not sure it is worth adding that complexity there.  What's the motivation to do this cleanup?

**Comment by jayunit100** [comment]
The motivation for this cleanup is because the `_PILLAR_` parameter crept into the local DNS docs, and I just happened to remember from 3 years ago where those came from.  Other folks were confused about wether they needed salt in order to run those recipes, so its creating some tech debt that now might creep into docs :).

**Comment by jayunit100** [comment]
@prameshj is there a subset of changes here that you'd consider safe? maybe that would be a good compromise as a start.

**Comment by prameshj** [comment]
> @prameshj is there a subset of changes here that you'd consider safe? maybe that would be a good compromise as a start.

Can we skip the changes to nodelocaldns yaml? The other config files(kubedns, coredns) don't have the same dependency.

**Comment by jayunit100** [comment]
okidoki

**Comment by jayunit100** [comment]
ok removed the changes to the node-local-dns.yaml stuff.  can maybe do that another time... if you have a suggestion of when would be a good idea lmk.

**Comment by jayunit100** [comment]
```release-note
NONE
```

File: cluster/addons/dns/kube-dns/kube-dns.yaml.in [hunk 2]
```diff
@@ -145,7 +145,7 @@ spec:
           initialDelaySeconds: 3
           timeoutSeconds: 5
         args:
-        - --domain={{ pillar['dns_domain'] }}.
+        - --domain=dns_domain
```
Comment (prameshj) [inline cluster/addons/dns/kube-dns/kube-dns.yaml.in:148]: There should be a trailing dot

Comment (jayunit100) [reply cluster/addons/dns/kube-dns/kube-dns.yaml.in]: ah gotcha , fixed

**Comment by jayunit100** [comment]
...  @johnbelamaric is this one all set. ? or any other issues

**Comment by jayunit100** [comment]
someone mentioned that the local-e2e was not a merge requirement but I haven't looked into it.... should I ?

**Comment by jayunit100** [comment]
```
error during make -C /go/src/k8s.io/kubernetes bazel-release: exit status 2 | error during make -C /go/src/k8s.io/kubernetes bazel-release: exit status 2
-- | --
error during make -C /go/src/k8s.io/kubernetes bazel-release: exit status 2
 
```
maybe Bazel related?

**Comment by jayunit100** [comment]
fixed, I was missing a `/` in the cluster/addons/dns/kube-dns/kube-dns.yaml.in file :) .

**Comment by jayunit100** [comment]
potential flakes:

```
[k8s.io] Container Runtime blackbox test when starting a container that exits should run with the expected status [NodeConformance] [Conformance] expand_less
```
and
```
[sig-scheduling] SchedulerPreemption [Serial] validates lower priority pod preemption by critical pod [Conformance] 
```

**Comment by jayunit100** [comment]
still assuming that `SchedulerPreemption` tests shouldn't be failing due to this change, so bumped them again.

**Comment by jayunit100** [comment]
scheduler pre-emption tests still failing here... no idea why :)

**Comment by jayunit100** [comment]
ok, i think i confirmed that this is just an artifact of the way the `test-infra` job runs for local-up-cluster.  

- The local-e2e test is doomed i guess to fail the SchedulerPreemption stuff since its a tiny one node cluster.
- I've made a PR there https://github.com/kubernetes/test-infra/pull/19142 to skip the `SchedulerPredicates` test when running local-e2e conformance.
- So, i guess the signal on this test is green,   and ready to merge now unless there are other concerns.

**Comment by jayunit100** [comment]
bump @johnbelamaric @prameshj the companion test fix for test-infra will merge soon, did you want to wait on this to get green local-e2e passing tests? or can we merge as is?

File: cluster/addons/dns/nodelocaldns/nodelocaldns.yaml (hunk 1 above)
Comment (prameshj) [reply cluster/addons/dns/nodelocaldns/nodelocaldns.yaml:99]: From the last discussion, we agreed to remove this diff from this PR, correct?

Comment (jayunit100) [reply cluster/addons/dns/nodelocaldns/nodelocaldns.yaml:99]: ah,

**Comment by jayunit100** [comment]
maybe a new flake?
```
E0910 17:36:29.534405  199314 remote_runtime.go:83] Version from runtime service failed: rpc error: code = Unavailable desc = transport is closing
176
    remote_runtime_test.go:60: 
177
        	Error Trace:	remote_runtime_test.go:60
178
        	Error:      	Received unexpected error:
179
        	            	rpc error: code = Unavailable desc = transport is closing
180
        	Test:       	TestVersion
181
--- FAIL: TestVersion (0.04s)
182
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
183
	panic: runtime error: invalid memory address or nil pointer dereference
```

**Comment by jayunit100** [comment]
cc @prameshj :) I think all set here?

**Comment by jayunit100** [comment]
cc @mikedanese and @aojea , follow up from https://github.com/kubernetes/kubernetes/pull/58248 about 2 years ago, theres still some remnants of salt in the codebase that are confusing the docs for  node local dns cache and so on.

**Comment by jayunit100** [comment]
BTW the failing test will be fixed I think once test-infra updates to using https://github.com/kubernetes/test-infra/pull/19142/files, which fails on local-up-cluster.  (also dims mentioned that the local-e2e ci job isn't actually a blocker)

**Comment by aojea** [comment]
> cc @mikedanese and @aojea , follow up from #58248 about 2 years ago, theres still some remnants of salt in the codebase that are confusing the docs for node local dns cache and so on.

I can' t help you with the salt bits sorry, that's an area that I never touched :)

File: cluster/addons/dns/kube-dns/transforms2salt.sed [hunk 3]
```diff
@@ -1,5 +1,5 @@
-s/__PILLAR__DNS__SERVER__/{{ pillar['dns_server'] }}/g
-s/__PILLAR__DNS__DOMAIN__/{{ pillar['dns_domain'] }}/g
-s/__PILLAR__CLUSTER_CIDR__/{{ pillar['service_cluster_ip_range'] }}/g
-s/__PILLAR__DNS__MEMORY__LIMIT__/{{ pillar['dns_memory_limit'] }}/g
+s/__DNS__SERVER__/{{ pillar['dns_server'] }}/g
+s/__DNS__DOMAIN__/{{ pillar['dns_domain'] }}/g
+s/__CLUSTER_CIDR__/{{ pillar['service_cluster_ip_range'] }}/g
+s/__DNS__MEMORY__LIMIT__/{{ pillar['dns_memory_limit'] }}/g
```
Comment (liggitt) [inline cluster/addons/dns/kube-dns/transforms2salt.sed:4]: this left pillar references in the file, but stopped replacing them in setup-kube-dns-manifest

Comment (aojea) [reply cluster/addons/dns/kube-dns/transforms2salt.sed:4]: I think that this issue was caught in the CI and fixed by https://github.com/kubernetes/kubernetes/pull/95580

Comment (liggitt) [reply cluster/addons/dns/kube-dns/transforms2salt.sed:4]: nope, setup-kube-dns-manifest is still wrong in master:

https://github.com/kubernetes/kubernetes/blob/c3c050c88f4265ba77464682801e2cc1ab8021f7/cluster/gce/gci/configure-helper.sh#L2492-L2495

Comment (prameshj) [reply cluster/addons/dns/kube-dns/transforms2salt.sed:4]: @liggitt the configure-helper.sh only looks at kube-dns.yaml.in which has no "pillar" references anymore.
https://github.com/kubernetes/kubernetes/blob/c3c050c88f4265ba77464682801e2cc1ab8021f7/cluster/addons/dns/kube-dns/kube-dns.yaml.in#L33

so the reference string replace above should be enough.
https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/dns/kube-dns/transforms2salt.sed still has some "pillar" references, but I don't think that is used anywhere.

 The issue @aojea is referring to was specific to nodelocaldns, which got fixed. 
Here is CI that is deploying kube-dns on a GCE k8s cluster.

**Pull request merged**